python/src/uagents/context.py

```diff
@@ -437,39 +459,39 @@ async def send_raw(
             sender=self.agent.address,
             target=destination_address,
             session=self._session,
-            # schema_digest=message_schema_digest,
-            protocol=message_schema_digest,  # why use 'protocol'?
+            schema_digest=message_schema_digest,
             protocol_digest=protocol_digest,
             expires=expires,
         )
         env.encode_payload(message_body)
         env.sign(self.agent.sign_digest)
 
-        self._queue_envelope(env, sync)
+        # Create awaitable future for sync messages responses
+        fut = asyncio.Future()
 
-        return MsgStatus(
-            status=DeliveryStatus.QUEUED,
-            detail="Envelope successfully queued",
-            destination=destination,
-            endpoint="",
-        )
-        # At this point we need the result of the send_raw_exchange_envelope
-        # but the result will be async so we need another way to access the
-        # return value
+        self._queue_envelope(env, endpoints, fut, sync)
 
-        # if isinstance(result, Envelope):
-        #     return await dispatch_sync_response_envelope(result)
+        result = await fut
```

Comment:
Should we use `asyncio.wait_for(fut, timeout)` here?

Reply:
Agree, I had it in earlier but reverted to the current state.
My reasoning was that the `send_exchange_envelope` method always returns a form of `MsgStatus` if something goes wrong but it wouldn't hurt to add a timeout fallback here.